app/internal/imagefetch/imagefetch.go

```
@@ -198,57 +306,161 @@ func PinnedReference(imageRef, digest string) (string, error) {
+// versa. Explicit input-limit failures always stop discovery and fail closed.
+func (f *Fetcher) visitSignatureImages(ctx context.Context, ref name.Reference, imageDigest v1.Hash, budget *candidateBudget, visit func(v1.Image) error) error {
 	seen := make(map[string]struct{})
-	var imgs []v1.Image
+	signatureImages := 0
 
-	add := func(img v1.Image) {
+	add := func(img v1.Image) error {
 		dig, err := img.Digest()
 		if err != nil {
-			return
+			return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 320) but it returns nil

(nilerr)